Realm/RLMSyncManager.h

```diff
@@ -22,6 +22,8 @@
 
 RLM_HEADER_AUDIT_BEGIN(nullability, sendability)
 
+// NEXT-MAJOR: This enum needs to be removed when access to the logger is removed
+// from the sync manager. 
```

Comment:
This type should presumably be deprecated?

Reply:
When we deprecate setting the log level and log function from the syncManager and we move it to be set  through the app or the realm. we should instead use `RLMLogLevel`.

Reply:
We're deprecating those in this PR, so we should also deprecate the types only used by those.